windowsforms/FlexGridShowcaseDemo/cs/MainForm.cs

```diff
@@ -0,0 +1,682 @@
+                    .ToArray();
+
+            dataTable.Clear();
+
+            var dataRows = dataTable.Rows;
+            for (var rowIndex = 0; rowIndex < rowCount; rowIndex++)
+            {
+                // Fill history data
+                var historyData = new int[12];
+                for (var i = 0; i < historyData.Length; i++)
+                {
+                    historyData[i] = _rnd.Next(0, 50);
+                };
+
+                dataRows.Add(
+                    // ID
+                    rowIndex + 1,
+                    // Product
+                    products[_rnd.Next(products.Length)],
+                    // Country
+                    countries[_rnd.Next(countries.Length)],
+                    // Color
+                    colors[_rnd.Next(colors.Length)],
+                    // Price
+                    _rnd.NextDouble() * 100 * _rnd.Next(100),
+                    // Change
+                    _rnd.NextDouble() * 10 * _rnd.Next(100) * (_rnd.Next(1, 3) == 1 ? (-1) : 1),
+                    // History
+                    historyData,
+                    // Discount
+                    _rnd.NextDouble(),
+                    // Rating
+                    _rnd.Next(0, 5),
+                    // Active
+                    (_rnd.Next(1, 3) == 1 ? false : true),
+                    // Date
+                    startPeriod.AddDays(_rnd.Next(60))
+                        .AddHours(_rnd.Next(60))
+                        .AddMinutes(_rnd.Next(60))
+                    );
+            }
+
+            _dataSet.AcceptChanges();
+            _flexGrid.EndUpdate();
+        }
+
+        private void FillData()
+        {
+            var descriptions = new List<string>()
+            {
+                "Across all our software products and services, our focus is on helping our customers achieve their goals. Our key principles – thoroughly understanding our customers' business objectives, maintaining a strong emphasis on quality, and adhering to the highest ethical standards – serve as the foundation for everything we do."
+            };
+
+            // Products table
+            var productsDataTable = new DataTable("Products");
+            
+            var productsColumns = productsDataTable.Columns;
+
+            // Add columns
+            productsColumns.Add("Name");
+            productsColumns.Add("Size", typeof(float));
+            productsColumns.Add("Weight", typeof(float));
+            productsColumns.Add("Quantity", typeof(uint));
+            productsColumns.Add("Description");
+
+            var productsRows = productsDataTable.Rows;
+
+            // Add rows
+            productsRows.Add("Gadget", 120f, 900f, 2, descriptions[_rnd.Next(descriptions.Count - 1)]);
+            productsRows.Add("Widget", 20f, 20f, 25, descriptions[_rnd.Next(descriptions.Count - 1)]);
+            productsRows.Add("Doohickey", 74f, 90f, 100, descriptions[_rnd.Next(descriptions.Count - 1)]);
```

Comment:
Is `- 1` intended here?
I feel it should be justdescriptions.Count

Reply:
Fixed with 381d8396ba274689199e36e08a7817ce1c84df05.